### DISCUSSION TITLE
At a loss on extended memory

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#125] [General]
Modval has the user's valind EC set up for unlimited. user extended memory is enabled after deadstart by hand, although the ipr decks seem to already do this. Other processes definitely show ecm in use, so I have to be missing something here...

**Comment by KC5VDJ** [comment]
Excuse the old batmud.fi bat, I use ASCII art to separate things with scrollback.

![XT63JMjicG_093755](https://github.com/kej715/DtCyber/assets/136194499/2199be77-e72b-4125-b90c-4c8240db47a3)

**Comment by kej715** [comment]
As ESM is used as a mass storage device, if you want to use it for user
extended memory also, you need to add an equipment deck entry, XM, defining
how much ESM is to be reserved for user extended memory. For example:

    XM=01,,400.

reserves 400K for user extended memory on the machine with ID "01".

Also, beware of enabling user extended memory with the 875 "beast"
configuration. Due to a bug in CPUMTR, the operating system will crash if
you enable user extended memory on this configuration and a user
attempts to allocate extended memory via the RFL command or underlying
system call. Apparently, CDC never actually built an 875 with a full
complement of ESM and also a full complement of CM, or user extended memory
usage was never tested on that configuration. In this configuration, when
CPUMTR attempts to initialize user extended memory, it incorrectly executes
an instruction that causes itself to be overwritten.
